### DISCUSSION TITLE
How to configure single pages shown before folders with "generateSite"

### DISCUSSION THREAD

**Discussion by micha-frey** [docToolchain/docToolchain#1095] [Q&A] [answered]
Hi all,

I want to show some single pages before the folders are shown or even mix them. Meaning, showing some single pages, then folders, and then pages again, followed by folders etc.

I have seen this on doctoolchain.org as well on the "news" section, where the overview is shown before the years' folders.
![image](https://user-images.githubusercontent.com/98160615/228595023-90ea9a18-f2e1-4cd4-b355-64320374f743.png)

Do you have any hints for me, please?

Thanks a lot and cool project :)

**Comment by rdmueller** [comment, marked as answer]
first, the news section uses a special template - it uses hte blog feature of jBake. As a result, it is not a good example for normal documentation.

but here is the solution to your question:

Create a folder structure like this one...

![image](https://user-images.githubusercontent.com/1856308/228649133-fe6c0f9c-fc53-48fe-b6e8-5d05e038fa1e.png)

it is currently important that you use the `_` and not `-` to separate the number from the file name. The number will determine the order of the elments. For an .adoc file, you can also use the `:jbake-order:` attribute to override this, but not for a folder.

This structure will then be rendered like to following screen:

![image](https://user-images.githubusercontent.com/1856308/228649038-88bb3473-76b6-4279-a177-6c61770574cc.png)

See also https://doctoolchain.org/docToolchain/v2.0.x/015_tasks/03_task_generateSite.html#_menu

Does this help?

**Reply by micha-frey** [reply]
Thanks for the fast and correct answer! Works as expected :)

**Reply by rdmueller** [reply]
just did a little change. It should now also be possible to use `-` as separator, e.g. `02-Sub-Page.adoc`
And you can map the name of a folder `02-Sub-Pages` to a menu entry by specifying it in the `menu` entry in the config as `["Sub-Pages": "a better name"]`
